godot-codegen/src/godot_version.rs

```diff
@@ -28,25 +29,44 @@ pub struct GodotVersion {
 
 pub fn parse_godot_version(version_str: &str) -> Result<GodotVersion, Box<dyn Error>> {
     let regex = Regex::new(
-        //  major  minor     [patch]                                              official|custom_build|gentoo
-        //  v      v         v                                                    v
-        r#"(\d+)\.(\d+)(?:\.(\d+))?\.(alpha|beta|dev|rc|stable)[0-9]*\.(?:mono\.)?(?:[a-z_]+\.([a-f0-9]+)|official)"#,
+        r#"(?x) # ignore whitespace and allow line comments (starting with `#`)
+        ^
+        (?P<major>\d+)
+        \.(?P<minor>\d+)
+        (?:\.(?P<patch>\d+))?
+        \.(?P<stability>alpha|beta|dev|rc|stable)[0-9]*
+        (\.[^.]+)+? # mono|official|custom_build|gentoo|arch_linux|...
+        (?:\.(?P<custom_rev>[a-f0-9]+))?
+        $
+    "#,
     )?;
 
     let fail = || format!("Version substring cannot be parsed: `{version_str}`");
     let caps = regex.captures(version_str).ok_or_else(fail)?;
 
     Ok(GodotVersion {
         full_string: caps.get(0).unwrap().as_str().to_string(),
-        major: caps.get(1).ok_or_else(fail)?.as_str().parse::<u8>()?,
-        minor: caps.get(2).ok_or_else(fail)?.as_str().parse::<u8>()?,
+        major: caps
+            .name("major")
+            .ok_or_else(fail)?
+            .as_str()
+            .parse::<u8>()?,
+        minor: caps
+            .name("minor")
+            .ok_or_else(fail)?
+            .as_str()
+            .parse::<u8>()?,
         patch: caps
-            .get(3)
+            .name("patch")
             .map(|m| m.as_str().parse::<u8>())
             .transpose()?
             .unwrap_or(0),
-        stability: caps.get(4).ok_or_else(fail)?.as_str().to_string(),
-        custom_rev: caps.get(5).map(|m| m.as_str().to_string()),
+        stability: caps
+            .name("stability")
+            .ok_or_else(fail)?
+            .as_str()
+            .to_string(),
+        custom_rev: caps.name("custom_rev").map(|m| m.as_str().to_string()),
     })
```

Comment:
A minor thing, but for these extractions, would it be possible to use a function/closure? 

Maybe something like 
```rs
fn read<T: FromStr>(captures: &Captures, key: &str) -> Result<T, Err>;
```
```rs
major: read(caps, "major")?,
minor: read(caps, "minor")?,
patch: read(caps, "patch").some(),
```

Reply:
Nice one. Done.